process/block/preprocess/rewardTxPreProcessor.go

```diff
@@ -339,21 +353,24 @@ func (rtp *rewardTxPreprocessor) setMissingTxsForShard(senderShardID uint32, mbT
 
 // computeMissingAndExistingRewardTxsForShards calculates what reward transactions are available and what are missing
 // from block.Body
-func (rtp *rewardTxPreprocessor) computeMissingAndExistingRewardTxsForShards(body block.Body) map[uint32][]*txsHashesInfo {
+func (rtp *rewardTxPreprocessor) computeMissingAndExistingRewardTxsForShards(body *block.Body) map[uint32][]*txsHashesInfo {
 	rewardTxs := block.Body{}
-	for _, mb := range body {
-		if mb.Type != block.RewardsBlock {
-			continue
-		}
-		if mb.SenderShardID == rtp.shardCoordinator.SelfId() {
-			continue
-		}
 
-		rewardTxs = append(rewardTxs, mb)
+	if body != nil {
```

Comment:
please use this construction instead:

if body == nil{
return make(map[uint32][]*txsHashesInfo)
}

...

Reply:
done